src/vendor/github.com/xelabs/go-mysqlstack/sqlparser/ddl_test.go

```diff
@@ -127,6 +127,56 @@ func TestDDL1(t *testing.T) {
 				"	`name` varchar(10)\n" +
 				") engine=tokudb default charset=utf8",
 		},
+		{
+			input: "create table t (\n" +
+				"	`status` int primary key,\n" +
+				"	`name` varchar(10)\n" +
+				") partition by hash(`status`)",
+			output: "create table t (\n" +
+				"	`status` int primary key,\n" +
+				"	`name` varchar(10)\n" +
+				")",
+		},
+		{
+			input: "create table t (\n" +
+				"	status int primary key comment '/*non_reserved_keyword*/',\n" +
+				"	`name` varchar(10)\n" +
+				") partition by hash(status)",
+			output: "create table t (\n" +
+				"	`status` int comment '/*non_reserved_keyword*/' primary key,\n" +
+				"	`name` varchar(10)\n" +
+				")",
+		},
+		{
+			input: "create table t (\n" +
+				"	bool int primary key comment '/*non_reserved_keyword*/',\n" +
+				"	`name` varchar(10)\n" +
+				") partition by hash(bool)",
+			output: "create table t (\n" +
+				"	`bool` int comment '/*non_reserved_keyword*/' primary key,\n" +
+				"	`name` varchar(10)\n" +
+				")",
+		},
+		{
+			input: "create table t (\n" +
+				"	enum int primary key comment '/*non_reserved_keyword*/',\n" +
+				"	`name` varchar(10)\n" +
+				") partition by hash(enum)",
+			output: "create table t (\n" +
+				"	`enum` int comment '/*non_reserved_keyword*/' primary key,\n" +
+				"	`name` varchar(10)\n" +
+				")",
+		},
+		{
+			input: "create table t (\n" +
+				"	datetime int primary key comment '/*non_reserved_keyword*/',\n" +
+				"	`name` varchar(10)\n" +
+				") partition by hash(datetime)",
+			output: "create table t (\n" +
+				"	`datetime` int comment '/*non_reserved_keyword*/' primary key,\n" +
+				"	`name` varchar(10)\n" +
+				")",
+		},
```

Comment:
Add some cases about moving some non_reserved _keywords to reserved_keywords

Reply:
Done. I have add some create fail cases about the non_reserved_keyword to reserved_keyword, as they are similar problems, I just picked some keywords here: bigitn, blob, char, decimal, integer.

Reply:
Just one or two is ok.